File: spec/php/ProcessXorValueTest.php
<?php
namespace Kaitai\Struct\Tests;

class ProcessXorValueTest extends TestCase {
    public function testProcessXorValue() {
        $r = ProcessXorValue::fromFile(self::SRC_DIR_PATH . "/process_xor_1.bin");

        $this->assertEquals(0xff, $r->key());
        $this->assertEquals("foo bar", $r->buf());
    }
}

File: spec/go/process_xor_value_test.go
package spec

import (
	"os"
	"testing"

	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	"github.com/stretchr/testify/assert"

	. "test_formats"
)

func TestProcessXorValue(t *testing.T) {
	f, err := os.Open("../../src/process_xor_1.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)

	var r ProcessXorValue
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, 0xff, r.Key)
	assert.EqualValues(t, "foo bar", r.Buf)
}
